lib/galaxy/tools/wrappers.py

```diff
@@ -83,7 +83,7 @@ def _get_cast_value(self, other):
         if self.input.type == 'boolean' and isinstance(other, string_types):
             return str(self)
         # For backward compatibility, allow `$wrapper != ""` for optional non-text param
-        if self.input.optional and self.value is None and isinstance(other, string_types):
+        if self.input.optional and self.value is None and self.input.type != 'boolean':
             return str(self)
```

Comment:
What about?
```python
        if self.input.optional and self.value is None:
            if isinstance(other, string_types):
                return str(self)
            else:
                return None
```

Reply:
Seems reasonable. At least, `None` seems to be expected by the bowtie2 and bwa-mem wrappers. The current fix only works sort of by chance because those wrappers do this with an optional integer param:

```
#if $rg_param("PI") != None
    #set $rg_pi = str($rg_param("PI"))
#else 
    #set $rg_pi = ''
#end if
```

Seems that today that whole logic is superfluous and could be replaced with just:
`#set $rg_pi = str($rg_param("PI"))`
but these wrappers seem to document that, historically, the expectation was that the comparison should be made against `None` not `''`.

Reply:
I'll give it a try.

Reply:
@wm75 Exactly, in 19.01 and earlier the bowtie2 code above would never reach the `else` branch because `$rg_param("PI")` is evaluated as an object.
With the recent changes in 19.05, the optional param is instead casted as `int(float(str(value)))`, which fails because `str(value)` evaluates to `""` and `float("")` raises the exception in the PR description.
I feel that evaluating an empty optional param to `None` makes more sense than `""`, except when comparing against a string, and can even simplify some convoluted syntax we have to use at the moment.

Reply:
I agree, and also ran the tests against tools-iuc, it seems we've now eliminated all cheetah errors (with or without the @nsoranzo's patch). I'll open a PR with the patch.